src/main/src/main/java/org/geoserver/catalog/impl/ModificationProxyCloner.java

```diff
@@ -194,7 +195,13 @@ public int compare(Class c1, Class c2) {
             // nothing to copy
             return null;
         }
-        Collection<T> copy = source.getClass().newInstance();
+        Collection<T> copy;
+        try {
+            copy = source.getClass().newInstance();
+        } catch (InstantiationException e) {
+            //we'll just pick something
+            copy = new ArrayList<T>();
```

Comment:
Thanks for the review. The problem is [this](https://github.com/NielsCharlier/geoserver/commit/e919df627299543e2177fc8f814f8a3dfce92f5b#diff-56150f92aa7ba936e390c7a4aa0438ffR50) inner class I made ( a list). I searched for a way to reliably clone any collection easily, but that doesn't seem to exist except if you use some third part library. But I guess the least I could do is add some extra ifs for sets etc... although that shouldn't occur (for now)